families/supplychain/restapi/openapi.yaml

```diff
@@ -0,0 +1,378 @@
+              data:
+                type: array
+                items:
+                  $ref: "#/definitions/Record"
+              link:
+                $ref: "#/definitions/Link"
+              paging:
+                $ref: "#/definitions/Paging"
+        400:
+          $ref: "#/responses/400BadRequest"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+  /records/{record_id}:
+    parameters:
+      - $ref: "#/parameters/record_id"
+    get:
+      summary: Fetches a particular Record.
+      responses:
+        200:
+          description: Successfully fetched Record
+          schema:
+            properties:
+              data:
+                $ref: "#/definitions/Record"
+              link:
+                $ref: "#/definitions/Link"
+        400:
+          $ref: "#/responses/400BadRequest"
+        404:
+          $ref: "#/responses/404NotFound"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+
+responses:
+  400BadRequest:
+    description: Request was malformed
+    schema:
+      $ref: "#/definitions/Error"
+  404NotFound:
+    description: Identifier did not match any resource
+    schema:
+      $ref: "#/definitions/Error"
+  500ServerError:
+    description: Something went wrong within the validator
+    schema:
+      $ref: "#/definitions/Error"
+  503ServiceUnavailable:
+    description: API is unable to reach the validator
+    schema:
+      $ref: "#/definitions/Error"
+
+parameters:
+  agent_id:
+    name: agent_id
+    in: path
+    type: string
+    required: true
+    description: Agent identifier, the public key the agent uses.
+  record_id:
+    name: record_id
+    in: path
+    type: string
+    required: true
+    description: "Record identifier, the serial number or natural identifier
+      of the item (material) being tracked."
+  count:
+    name: count
+    in: query
+    type: integer
+    default: 1000
+    description: Number of items to return
+  min:
+    name: min
+    in: query
+    type: string
+    description: Id or index to start paging (inclusive)
+  max:
+    name: max
+    in: query
+    type: string
+    description: Id or index to end paging (inclusive)
+
+definitions:
+  Link:
+    type: string
+    example: https://api.sawtooth.com/agents/65cd3a3ce088b265b626f704b7f3db97b6f12e848dccb35d7806f3d0324c71b709ed360d602b8b658b94695374717e3bdb4b76f77886953777d5d008558247dd
+  Paging:
+    properties:
+      start_index:
+        type: integer
+        example: 1000
+      total_count:
+        type: integer
+        example: 54321
+      previous:
+        type: string
+        example: https://api.sawtooth.com/agents?min=0&count=1000
+      next:
+        type: string
+        example: https://api.sawtooth.com/agents?min=2000&count=1000
+  Error:
+    properties:
+      code:
+        type: integer
+        example: 34
```

Comment:
My comment is on the entire example error, not just the error code. The examples should all be something that might plausibly get returned. If you use an existing error code, for an error that does not apply to your API, then we know this example will never actually be returned. I was suggesting using an existing error your API would be likely to use.

As to your other question, a _user_ would not be interested in an error code any more than they are interested in a `404` status code, but a _developer_ might be, especially if they plan to write code around it. I don't expect them to read the python source code, but the official documentation here:

http://intelledger.github.io/rest_api/error_codes.html